adapters/fwssp/fwssp.go

```
@@ -0,0 +1,115 @@
+		imp := &request.Imp[i]
+		var bidderExt adapters.ExtImpBidder
+		if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			return nil, []error{&errortypes.BadInput{
+				Message: fmt.Sprintf("Invalid imp.ext for impression index %d. Error Infomation: %s", i, err.Error()),
+			}}
+		}
+
+		var impExt openrtb_ext.ImpExtFWSSP
+		if err := jsonutil.Unmarshal(bidderExt.Bidder, &impExt); err != nil {
+			return nil, []error{&errortypes.BadInput{
+				Message: fmt.Sprintf("Invalid imp.ext for impression index %d. Error Infomation: %s", i, err.Error()),
+			}}
+		}
+
+		var err error
+		if imp.Ext, err = jsonutil.Marshal(impExt); err != nil {
+			return nil, []error{&errortypes.BadInput{
+				Message: fmt.Sprintf("Unable to transfer requestImpExt to Json fomat, %s", err.Error()),
+			}}
+		}
+	}
+
+	requestJSON, err := jsonutil.Marshal(request)
+	if err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unable to transfer request to Json fomat, %s", err.Error()),
+		}}
+	}
+
+	headers := http.Header{}
+	headers.Add("Componentid", "prebid-go")
+
+	requestData := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    requestJSON,
+		Headers: headers,
+		ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+	}
+	return []*adapters.RequestData{requestData}, nil
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent ||
+		(response.StatusCode == http.StatusOK && len(response.Body) == 0) {
+		return nil, nil
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+
+	var bidResp openrtb2.BidResponse
+	if err := jsonutil.Unmarshal(response.Body, &bidResp); err != nil {
```

Please add a supplemental JSON test where your mock response returns a bid response that fails to unmarshal successfully. This can be achieved by setting one of the fields to an incorrect data type.

---

added